var/spack/repos/builtin/packages/neurodamus/package.py

```diff
@@ -108,25 +90,26 @@ def build(self, spec, prefix):
 
         if '+syntool' in spec:
             include_flag += ' -DENABLE_SYNTOOL -I ' + spec['synapsetool'].prefix.include
-            link_flag += self.syntool_dep_libs()
+            link_flag += spec['synapsetool'].libs.rpath_flags
+            link_flag += ' ' + spec['synapsetool'].libs.ld_flags
 
         if '+coreneuron' in spec:
             include_flag += ' -DENABLE_CORENEURON -I%s' % (spec['coreneuron'].prefix.include)
-            link_flag += ' %s' % (spec['coreneuron'].libs.ld_flags)
+            link_flag += ' ' + spec['coreneuron'].libs.rpath_flags
+            link_flag += ' ' + spec['coreneuron'].libs.ld_flags
 
         include_flag += ' -I%s -I%s %s' % (spec['reportinglib'].prefix.include,
                                            spec['hdf5'].prefix.include,
                                            profile_flag)
-        link_flag += ' %s -L%s -lhdf5 -L%s -lz' % (
+        link_flag += ' %s %s -L%s -lhdf5 -L%s -lz' % (
+                     spec['reportinglib'].libs.rpath_flags,
                      spec['reportinglib'].libs.ld_flags,
                      spec['hdf5'].prefix.lib,
                      spec['zlib'].prefix.lib)
```

Comment:
>The rpath_flags wasnt to be used with synapsetool/reporting lib only when they satisfy +shared?

yes, its true...but I didn't add it to avoid if condition (adding rpath doesn't hurt anyway)

> Btw zlib has been a dep since the beginning, but to be sure, we have mods using zlib directly?

thats hdf5 dependency...if we link with hdf5, we have to also need to link with zlib and hence that is added as dependency.

Reply:
I did some tests and shared libs have RPath, so actually we avoid all problems of finding sub-dependencies and things not working when using external packages if we just look for direct dependencies. 
We only need that if we want to support building on static libraries